### ISSUE TITLE
WebDriverAgent runner installs in a repeated loop on iOS 17.3 

### ISSUE THREAD

**Issue by snWoody** [shamanec/GADS-devices-provider#44]
Hi, I have followed the instructions given in GADS device provider & GADS UI. When I try to launch the UI, i see that the WebDriverAgent runner is getting installed repeatedly in a loop on the device. When I look at the logs, it shows an error: 
"Error waiting for WebDriverAgent(xcodebuild) command to finish.. it is errored out or device <> disconnected agent killed"

I am not sure what is wrong with the setup. Can you advise?
Env -> iOS 17.3 and mac os ventura 13.6.4

**Comment by shamanec** [comment]
Lol, I've never seen `agent killed` error before :sweat_smile: Ventura means you are not using Xcode 15? I would assume this is the reason for iOS 17 failures on first glance. Have you set up the signing for WebDriverAgent? Did you try running it on a device through Xcode directly?

**Comment by snWoody** [comment]
I am using  Xcode build15.2  and signing profile and other things have taken care.  Here is something I have observed 
- when I run xcodebuild test-without-building command via terminal . I see that webdriver agent runner is working as expected and http end point deviceip/status returns response 
- But when I run xcodebuild test-with-building command installs webdriveragentrunner repeatedly in a loop.
- I tried to modify one of the files iOS.go , line 76 . It did not help.
I assume webdriveragent.ipa file is not needed for ios. 
Can you advise?

**Comment by shamanec** [comment]
The `build-for-testing` command is executed only once on provider startup, this is to speed up deployment on each device later with `test-without-building`. In general at the moment `*.ipa` file is not needed because the provider uses `xcodebuild` directly to deploy on the devices from the already built WebDriverAgent code. Do you have paid developer account? Also I was asking if you tried running WebDriverAgent from Xcode directly, maybe the developer needs to be trusted from the device settings.

**Comment by shamanec** [comment]
I would suggest you try out the `develop` branches for both the provider and the UI. There are a few differences in setup and you'll have to drop the current MongoDB data, but in theory it should work better

**Comment by snWoody** [comment]
I have got a Developer account. It worked when I ran directly with Xcode . The build was successful, trusted the developer on the device.. I can see the `<deviceip>/status` endpoint returns valid response.
Let me try with develop branch.

Thank you!

**Comment by shamanec** [comment]
I've created a Discord server that is in the `develop` branch readme, you can join if you want so we can chat/debug faster :P
